backend/Techfesia2019/tickets/views.py

```diff
@@ -219,3 +232,50 @@ def put(self, request, public_id, format=None):
         ticket.subscribers.remove(profile)
         ticket.save()
         return Response({'message': 'unsubscribed from ticket'}, status=status.HTTP_200_OK)
+
+@receiver(notifications)
+def send_emails(sender, **kwargs):
+    #When tickets are closed
+    if(sender == Ticket):
+        ticket_id = kwargs['id']
+        ticket = Ticket.objects.get(public_id = ticket_id)
+        message = 'Ticket has been closed by {}'.format(ticket.solved_by.user.username)
+        context = {
+            'ticket' : ticket,
+            'message': message,
+            'last_comment': None,
+            'url': 'localhost:8000/ticket/{}/detail'.format(ticket_id)
+        }
+        subject = 'Ticket no.{} is solved'.format(ticket_id)
+    #When new comments are created
+    else:
+        comment_id = kwargs['id']
+        comment = TicketComment.objects.get(public_id = comment_id)
+        ticket = comment.ticket
+        message =  'A new comment has been posted by {}'.format(comment.commenter.user.username)
+        context = {
+            'ticket' : ticket,
+            'message': message,
+            'last_comment': comment,
+            'url': 'localhost:8000/ticket/{}/detail'.format(comment.ticket.public_id)
+        }
+        subject = 'Ticket no.{} has a new comment'.format(comment.ticket.public_id)
+    
+    from_email = settings.EMAIL_HOST_USER;
+    to_email = []   
+    #Get all the subscribers email_ids
+    for subscriber in ticket.subscribers.all():
+        to_email.append(subscriber.user.email)
+
+    html_message = get_template('tickets/base.html').render(context)
+    text_message = message
+    
```

Comment:
I haven't used this method before. Please ensure that it is also working with sendgrid. You can use my API Key or create your own account. On github student liscence you can get 50k emails per month